src/Tools/perf/ExternalProcessExecutor.cs

```diff
@@ -0,0 +1,104 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using BenchmarkDotNet.Engines;
+using BenchmarkDotNet.Environments;
+using BenchmarkDotNet.Horology;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Toolchains;
+using BenchmarkDotNet.Toolchains.Parameters;
+using BenchmarkDotNet.Toolchains.Results;
+
+namespace perf
+{
+    /// <summary>
+    /// Executor designed to take a Benchmark that lists a target executable
+    /// and a set of command line parameters and executes that process using
+    /// an external dotnet runtime process.
+    /// </summary>
+    internal class ExternalProcessExecutor : IExecutor
+    {
+        public ExecuteResult Execute(ExecuteParameters executeParameters)
+        {
+            if (!(executeParameters.Benchmark is ExternalProcessBenchmark))
+            {
+                throw new ArgumentException("Benchmark given is not an ExternalProcessBenchmark");
+            }
+
+            var exePath = executeParameters.BuildResult.ArtifactsPaths.ExecutablePath;
+
+            var benchmark = (ExternalProcessBenchmark)executeParameters.Benchmark;
+
+            using (var proc = new Process { StartInfo = CreateStartInfo(exePath, benchmark) })
+            {
+                return Execute(proc, benchmark, executeParameters.Logger);
+            }
+        }
+
+        private ExecuteResult Execute(Process proc, ExternalProcessBenchmark benchmark, ILogger logger)
+        {
+            logger.WriteLineInfo("// Execute: " + proc.StartInfo.FileName + " " + proc.StartInfo.Arguments);
+
+            var invokeCount = benchmark.Job.Run.TargetCount;
+            var measurements = new string[invokeCount];
+
+            for (int i = 0; i < invokeCount; i++)
+            {
+                var clock = Chronometer.BestClock;
+                var start = clock.Start();
+
+                proc.Start();
+                proc.WaitForExit();
+
+                var span = start.Stop();
+
+                var measurement = new Measurement(0, IterationMode.Result, i, 1, span.GetNanoseconds()).ToOutputLine();
+                Console.WriteLine(measurement);
+                measurements[i] = measurement;
+            }
+
+            return new ExecuteResult(true, proc.ExitCode, measurements, Array.Empty<string>());
+        }
+
+        private ProcessStartInfo CreateStartInfo(string exePath, ExternalProcessBenchmark benchmark)
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                WorkingDirectory = benchmark.WorkingDirectory,
+                RedirectStandardError = true,
```

Comment:
Is something reading from these streams? Note if the buffer gets full and nothing is reading then .WaitForExit() can deadlock.